web-app/HeimScripts/heatmap/summary.R

```diff
@@ -29,14 +29,43 @@
 library(jsonlite)
 library(gplots)
 
-main <- function(phase)
+main <- function(phase = NULL)
 {
+  check_input(loaded_variables, phase)
   data_measurements <- extract_measurements(loaded_variables)
-  produce_summary_stats(data_measurements, phase)
+  summary_stats_json <- produce_summary_stats(data_measurements, phase)
+  write_summary_stats(summary_stats_json)
   produce_boxplot(data_measurements, phase)
   return(list(summary_stats = "Finished")) #right now a non-empty list is expected as a return.
 }
 
+#check if provided variables and phase info are in line with expected input as described at top of this script
+check_input <- function(datasets, phase_info)
+{
+  #expected input: list of data.frames
+  items_list <- sapply(datasets, class)
+  if(class(datasets) != "list" | !all(items_list == "data.frame")) #for a data.frame is.list() also returns TRUE. Class returns "data.frame" in that case
+  { 
+    stop("Unexpected input. Expected input: a list, containing one or more data.frames")
+  }
+   
+  # all items in the list are expected to have some unique identifier for the node (numerical identifier appended behind the letter "n") 
+  # followed an underscore and a subset identifier s1 or s2 depending on subset, e.g. n0_s1, n0_s2, n1_s1, n1_s2. 
+  dataset_names <- names(datasets)
+  expected_format_names <- "^n[[:digit:]]+_s[[:digit:]]+$"
+  names_in_correct_format <- grepl(expected_format_names, dataset_names)
+  if(any(!names_in_correct_format))
+  {
+    stop(paste("One or more labels of the datasets do not have the expected format.", 
+               "\nExpected format: an unique numerical identifier for the node appended behind the letter \"n\",\nfollowed by an underscore and an unique numerical identifier for the subset appended behind an \"s\",",
+               "\ne.g. n0_s1, n0_s2, n1_s1, n1_s2. "))
+  }
+  if(is.null(phase_info))
+  {
+    stop("supply phase parameter to function \'main()\'")
```

Comment:
If we use stop - Frontend will just return unspecific error, without any meaningful message for the user. 
Would it not be better to : 
- set messages variable with appropriate message for the user
- do not call stop, let the script return (just without the boxplot of course)